src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts

```diff
@@ -924,349 +924,358 @@ const editorOptionsCustomizationSchema: IConfigurationPropertySchema = {
-			markdownDescription: nls.localize('notebook.outputFontFamily', "The font family of the output text within notebook cells. When set to empty, the {0} is used.", '`#editor.fontFamily#`'),
-			type: 'string',
-			tags: ['notebookLayout', 'notebookOutputLayout']
-		},
-		[NotebookSetting.outputScrolling]: {
-			markdownDescription: nls.localize('notebook.outputScrolling', "Initially render notebook outputs in a scrollable region when longer than the limit."),
-			type: 'boolean',
-			tags: ['notebookLayout', 'notebookOutputLayout'],
-			default: typeof product.quality === 'string' && product.quality !== 'stable' // only enable as default in insiders
-		},
-		[NotebookSetting.outputWordWrap]: {
-			markdownDescription: nls.localize('notebook.outputWordWrap', "Controls whether the lines in output should wrap."),
-			type: 'boolean',
-			tags: ['notebookLayout', 'notebookOutputLayout'],
-			default: false
-		},
-		[NotebookSetting.defaultFormatter]: {
-			description: nls.localize('notebookFormatter.default', "Defines a default notebook formatter which takes precedence over all other formatter settings. Must be the identifier of an extension contributing a formatter."),
-			type: ['string', 'null'],
-			default: null,
-			enum: DefaultFormatter.extensionIds,
-			enumItemLabels: DefaultFormatter.extensionItemLabels,
-			markdownEnumDescriptions: DefaultFormatter.extensionDescriptions
-		},
-		[NotebookSetting.formatOnSave]: {
-			markdownDescription: nls.localize('notebook.formatOnSave', "Format a notebook on save. A formatter must be available and the editor must not be shutting down. When {0} is set to `afterDelay`, the file will only be formatted when saved explicitly.", '`#files.autoSave#`'),
-			type: 'boolean',
-			tags: ['notebookLayout'],
-			default: false
-		},
-		[NotebookSetting.insertFinalNewline]: {
-			markdownDescription: nls.localize('notebook.insertFinalNewline', "When enabled, insert a final new line into the end of code cells when saving a notebook."),
-			type: 'boolean',
-			tags: ['notebookLayout'],
-			default: false
-		},
-		[NotebookSetting.formatOnCellExecution]: {
-			markdownDescription: nls.localize('notebook.formatOnCellExecution', "Format a notebook cell upon execution. A formatter must be available."),
-			type: 'boolean',
-			default: false
-		},
-		[NotebookSetting.confirmDeleteRunningCell]: {
-			markdownDescription: nls.localize('notebook.confirmDeleteRunningCell', "Control whether a confirmation prompt is required to delete a running cell."),
-			type: 'boolean',
-			default: true
-		},
-		[NotebookSetting.findFilters]: {
-			markdownDescription: nls.localize('notebook.findFilters', "Customize the Find Widget behavior for searching within notebook cells. When both markup source and markup preview are enabled, the Find Widget will search either the source code or preview based on the current state of the cell."),
+
+class NotebookSettingContribution implements IWorkbenchContribution {
```

Comment:
Please diff with white spaces ignored.
Adding this class to add a new setting for the notebook diffing algorithm that can be turned off.
Id like to add a new setting for notebook diffing algorithm (one used to better identify inserted/deleted cells).
& have users turn that off if not required (as a fall back).

I would also like to enable the setting for insiders by default.
However that requires the use of `IProductService`, hence the class.

@rebornix /cc